File: Android/app/src/main/java/com/edu/primary/utils/Logger.java
package com.edu.primary.utils;

import android.util.Log;
import com.edu.primary.BuildConfig;

/**
 * 日志工具类（统一日志管理）
 * 提供统一的日志记录接口，便于调试和错误追踪
 * Release版本自动禁用调试日志
 */
public class Logger {
    
    private static final String TAG = "GoodGoodStudy";
    // Release版本自动禁用调试日志，减少性能开销
    private static final boolean DEBUG = BuildConfig.DEBUG;
    
    private Logger() {
        // 工具类，禁止实例化
    }
    
    /**
     * 记录调试信息
     */
    public static void d(String message) {
        if (DEBUG) {
            Log.d(TAG, message);
        }
    }
    
    /**
     * 记录调试信息（带标签）
     */
    public static void d(String tag, String message) {
        if (DEBUG) {
            Log.d(TAG, "[" + tag + "] " + message);
        }
    }
    
    /**
     * 记录信息
     */
    public static void i(String message) {
        Log.i(TAG, message);
    }
    
    /**
     * 记录信息（带标签）
     */
    public static void i(String tag, String message) {
        Log.i(TAG, "[" + tag + "] " + message);
    }
    
    /**
     * 记录警告
     */
    public static void w(String message) {
        Log.w(TAG, message);
    }
    
    /**
     * 记录警告（带标签）
     */
    public static void w(String tag, String message) {
        Log.w(TAG, "[" + tag + "] " + message);
    }
    
    /**
     * 记录错误
     */
    public static void e(String message) {
        Log.e(TAG, message);
    }
    
    /**
     * 记录错误（带标签和异常）
     */
    public static void e(String tag, String message, Throwable throwable) {
        Log.e(TAG, "[" + tag + "] " + message, throwable);
    }
    
    /**
     * 记录错误（带异常）
     */
    public static void e(String message, Throwable throwable) {
        Log.e(TAG, message, throwable);
    }
}

File: Android-1/app/src/main/java/com/edu/primary/util/Logger.kt
package com.edu.primary.util

import android.util.Log
import com.edu.primary.BuildConfig

private const val TAG = "GoodGoodStudy"

/**
 * 统一日志封装，DEBUG 下输出 d，w/e 始终输出；所有日志带统一 TAG 与子 tag。
 */
object Logger {
    /** Debug 日志，仅 DEBUG 构建时输出。 */
    fun d(tag: String, msg: String) {
        if (BuildConfig.DEBUG) Log.d(TAG, "[$tag] $msg")
    }
    /** Warning 日志。 */
    fun w(tag: String, msg: String) = Log.w(TAG, "[$tag] $msg")
    /** Error 日志，可选附带异常。 */
    fun e(tag: String, msg: String, t: Throwable? = null) {
        if (t != null) Log.e(TAG, "[$tag] $msg", t) else Log.e(TAG, "[$tag] $msg")
    }
}
